src/uu/date/src/date.rs

```diff
@@ -243,6 +244,14 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                 Ok(date) => {
                     // GNU `date` uses `%N` for nano seconds, however crate::chrono uses `%f`
                     let format_string = &format_string.replace("%N", "%f");
+                    //Hack to work around panic in chrono,
+                    //TODO - removed when a fix for https://github.com/chronotope/chrono/issues/623 is released
+                    if StrftimeItems::new(format_string).any(|i| i == Item::Error) {
+                        return Err(USimpleError::new(
+                            1,
+                            format!("invalid format {}", format_string.replace("%f", "%N")),
+                        ));
+                    }
                     let formatted = date.format(format_string).to_string().replace("%f", "%N");
```

Comment:
I think we can use `format_with_items` here to avoid parsing the format string twice: https://docs.rs/chrono/latest/chrono/struct.Date.html#method.format_with_items.

Maybe we can even parse the string and do the error checking before the loop and reuse it.

Reply:
Oh and could you put a space after the `//` for the comments? It just a bit nicer :)